### PR TITLE
[Intel MKL]: Fixing AVX Performance Issue

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**
This PR fixes the MKL performance on machines with AVX instruction set by using some functionality from MKL binary version. Also, a new config option was added. If that option is used, only MKL-DNN open source lib will be used, without depending on MKL binary version.

### CODE REVIEW

**Review by gunan — DISMISSED** [review]

**Comment by martinwicke** [comment]
Can you look at the build failures?

**Comment by mahmoud-abuzaina** [comment]
@martinwicke sorry, there was a file missing from this PR. Can you re-run the test?

**Comment by martinwicke** [comment]
I think you have to export build_defs.bzl in your BUILD file?

**Comment by mahmoud-abuzaina** [comment]
@martinwicke I did.

**Comment by martinwicke** [comment]
Ah, never mind. The license check failed. It is there to ensure that for each third_party package we depend on which requires we include the license in the distribution, we actually include the license in the distribution. MKL-DNN may have a license that falls in that category? In that case, the fix would be to include the license in the deps for the pip package.

@yifeif can you correct me or fill in details?

**Comment by yifeif** [comment]
Yea we include the licenses for the third party packages we depends on.
`Missing the licenses for the following external dependencies:
//third_party/mkl_dnn`
Mind adding the license to the following targets? https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/pip_package/BUILD#L123
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/lib_package/BUILD#L111
and
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/lib_package/BUILD#L153

**Comment by mahmoud-abuzaina** [comment]
Thanks for the info @martinwicke and @yifeif . I made the change.

**Comment by mahmoud-abuzaina** [comment]
@yifeif it is still failing after adding the license. I am confused why the error message says "Please remove the licenses for the following external dependencies:
@mkl_dnn//". Any idea?

File: tensorflow/tools/lib_package/BUILD [hunk 1]
```diff
@@ -182,6 +183,7 @@ genrule(
         "@zlib_archive//:zlib.h",
     ] + if_mkl([
         "//third_party/mkl:LICENSE",
+        "@mkl_dnn//:LICENSE",
```
Comment (yifeif) [inline tensorflow/tools/lib_package/BUILD]: Should it be `"//third_party/mkl_dnn:LICENSE"`?

**Comment by yifeif** [comment]
Thanks for the commit @mahmoud-abuzaina. Left a comment inline.

File: tensorflow/tools/lib_package/BUILD (hunk 1 above)
Comment (mahmoud-abuzaina) [reply tensorflow/tools/lib_package/BUILD]: but I can try "//third_party/mkl_dnn:LICENSE" and see.

**Comment by mahmoud-abuzaina** [comment]
@martinwicke the tests seem to be passing now. Can we merge it?

**Review by martinwicke — APPROVED** [review]

**Comment by martinwicke** [comment]
@yifeif is copybara import stuck? It's been a long time since the label was applied...

**Comment by mahmoud-abuzaina** [comment]
@martinwicke it looks like all tests are passing now. Maybe we can merge this.

**Comment by yifeif** [comment]
hi @mahmoud-abuzaina sorry for the delay, I'm working on bring this change in internally. Once it gets submitted, it will get merged. Will ping back if anything comes up. Thanks!

**Comment by mahmoud-abuzaina** [comment]
Thank you!

**Pull request merged**